### PR TITLE
chore: refactor redirect URIs component into generic text field array component

### PR DESCRIPTION
**Pull request by Laurin-W**

### CODE REVIEW

File: src/components/VerboseTextFieldArray.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  onChange?:
+    | ChangeEventHandler<HTMLInputElement | HTMLTextAreaElement>
+    | undefined;
+  onBlur?:
+    | React.FocusEventHandler<HTMLInputElement | HTMLTextAreaElement>
+    | undefined;
+  pushItem(obj: unknown): void;
+  removeItem<T>(index: number): T | undefined;
+};
```
Comment (NSeydoux) [inline src/components/VerboseTextFieldArray.tsx:45]: Is it necessary to define this type entirely from scratch ? Or would have it been possible to reuse and extend the existing `TextFieldProps`, just adding the additional fields you are defining ?

File: src/components/VerboseTextFieldArray.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+export default function VerboseTextFieldArray(
+  props: VerboseFieldArrayRenderProps | void
```
Comment (NSeydoux) [inline src/components/VerboseTextFieldArray.tsx:48]:
```suggestion
  props?: VerboseFieldArrayRenderProps
```

File: src/components/VerboseTextFieldArray.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                  inputProps={{ inputMode: "url" }}
+                  required
+                  fullWidth
+                  size="small"
+                />
+              </Grid>
+              <Grid item>
+                <Button
+                  type="button"
+                  title="Remove Redirect URI of this row"
```
Comment (NSeydoux) [inline src/components/VerboseTextFieldArray.tsx:101]: This looks very specific to a certain context, which is somewhat misaligned with the component name which suggests it is generic and reusable. Should this be passed down as a prop too?

**Review by NSeydoux — COMMENTED** [review]
Just a couple of questions, but overall this looks good!

File: src/components/VerboseTextFieldArray.tsx (hunk 1 above)
Comment (Laurin-W) [reply src/components/VerboseTextFieldArray.tsx:45]: 👍🏼 Changed to `TextFieldProps & {...}`

File: src/components/VerboseTextFieldArray.tsx (hunk 3 above)
Comment (Laurin-W) [reply src/components/VerboseTextFieldArray.tsx:101]: Ah snap! That was not supposed to be there :D

**Review by NSeydoux — DISMISSED** [review]

**Review by NSeydoux — APPROVED** [review]

**Pull request merged**